connector/github/github.go

```diff
@@ -575,7 +580,21 @@ func (c *githubConnector) userEmail(ctx context.Context, client *http.Client) (s
 			}
 
 			if email.Verified && email.Primary {
-				return email.Email, nil
+				primaryEmail = email
+			}
+
+			if c.preferredEmailDomain != "" {
+				components := strings.Split(email.Email, "@")
+				if email.Verified && components[1] == c.preferredEmailDomain {
```

Comment:
NIT: WDYT about making preferredEmailDomain a glob so that users can provide a pattern (not the exact domain)?